macros/quick_are_queries_identical.sql

```diff
@@ -0,0 +1,51 @@
+/*
+As described by the Infinite Lambda team here: https://infinitelambda.com/data-validation-refactoring-snowflake/
+
+Some platforms let you take a hash of the whole table, which can be very very fast compared to comparing each row. 
+
+If you run this and it returns false, you still have to run the more in-depth queries to find out what specific changes there are, 
+but it's a good way to quickly verify identical results if that's what you're expecting. 
+*/
+
+{% macro quick_are_queries_identical(query_a, query_b, columns=[], event_time=None) %}
+    {{ return (adapter.dispatch('quick_are_queries_identical', 'audit_helper')(query_a, query_b, columns, event_time)) }}
+{% endmacro %}
+
+{% macro default__quick_are_queries_identical(query_a, query_b, columns, event_time) %}
+    {% set joined_cols = columns | join(", ") %}
+    {% if event_time %}
+        {% set event_time_props = audit_helper.get_comparison_bounds(a_query, b_query, event_time) %}
+    {% endif %}
+
+    select count(distinct hash_result) = 1 as are_tables_identical
+    from (
+        select hash_agg({{ joined_cols }}) as hash_result
+        from ({{ query_a }}) query_a_subq
+        {% if event_time_props %}
+            where {{ event_time_props["event_time"] }} >= '{{ event_time_props["min_event_time"] }}'
+            and {{ event_time_props["event_time"] }} <= '{{ event_time_props["max_event_time"] }}'
+        {% endif %}
+
+        union all
+        
+        select hash_agg({{ joined_cols }}) as hash_result
+        from ({{ query_b }}) query_b_subq
+        {% if event_time_props %}
+            where {{ event_time_props["event_time"] }} >= '{{ event_time_props["min_event_time"] }}'
+            and {{ event_time_props["event_time"] }} <= '{{ event_time_props["max_event_time"] }}'
+        {% endif %}
+
+    ) as hashes
+{% endmacro %}
+
+{% macro is_quick_are_queries_identical_supported() %}
+    {{ return (adapter.dispatch('is_quick_are_queries_identical_supported', 'audit_helper')()) }}
+{% endmacro %}
```

Comment:
This is not called anywhere.

Is it supposed to be called on its own?
Or, should we add it to `default__quick_are_queries_identical` and raise an exception when we are running a query on a non supported adapter?

Reply:
Also wondering this! My preference would be the latter to avoid _requiring_ the caller to make a call to `is_quick_are_queries_identical_supported` to safely call `quick_are_queries_identical`.

Reply:
Good shout! Will do the latter 👍